core/app/models/spree/promotion_handler/promotion_duplicator.rb

```diff
@@ -9,8 +9,7 @@ def initialize(promotion, random_string: nil)
       def duplicate
         @new_promotion = @promotion.dup
         @new_promotion.path = "#{@promotion.path}_#{@random_string}"
-        @new_promotion.name = "New #{@promotion.name}"
-        @new_promotion.code = "#{@promotion.code}_#{@random_string}"
+        @new_promotion.generate_code=(true)
```

Comment:
Could you explain the effect of these changes? This is an existing class that can be used by existing Spree users, it would be great to minimize the impact on them.

Also, what's this syntax `=(true)`?

Reply:
I thought it might be the case but hoped to reuse this. Thanks for pointing this out - I'll revert these changes so there's no impact on existing Spree users, and introduce the change in a more suitable place.

As for the syntax - it points to the `Promotion`'s setter method here - https://github.com/spree/spree/blob/main/core/app/models/spree/promotion.rb#L68